api/envoy/api/v2/core/base.proto

```diff
@@ -78,10 +79,48 @@ message Locality {
   string sub_zone = 3;
 }
 
+// BuildVersion combines SemVer version of extension with free-form build information
+// (i.e. 'alpha', 'private-build') as a set of strings.
+message BuildVersion {
+  // SemVer version of extension.
+  type.SemanticVersion version = 1;
+
+  // Free-form build information.
+  // Envoy defines several well known keys in the source/common/common/version.h file
+  google.protobuf.Struct metadata = 2;
+}
+
+// Version and identification for an Envoy extension.
+// [#next-free-field: 6]
+message Extension {
+  // This is the name of the Envoy filter as specified in the Envoy
+  // configuration, e.g. envoy.router, com.acme.widget.
+  string name = 1;
+
+  // Category of the extension.
+  // Extension category is provided by the `category()` method of the factory class.
+  // [#comment:TODO(yanavlasov): Link to the doc with existing category names.]
```

Comment:
I agree with @markdroth point, we should be using reverse DNS for these categories. Can we ensure that all the existing categories have a `envoy.` prefix and that we recommend folks use reverse DNS for custom categories?